tsod/detectors.py

```diff
@@ -228,30 +228,60 @@ class ConstantValueDetector(Detector):
 
     def __init__(self, window_size: int = 3, threshold: float = 1e-7):
         super().__init__()
+        
+        # Validate input
+        assert threshold >= 0, f"threshold must be non-negative, got {threshold}"
+        assert window_size >= 2, f"window_size must be at least 2, got {window_size}"
+        
         self._threshold = threshold
         self._window_size = window_size
+    
+    @property
+    def threshold(self) -> float:
+        return self._threshold
+
+    @property
+    def window_size(self) -> int:
+        return self._window_size
 
     def _fit(self, data):
         return self
 
-    def _detect(self, data: pd.Series) -> pd.Series:
-        rollmax = data.rolling(self._window_size, center=True).apply(np.nanmax)
-        rollmin = data.rolling(self._window_size, center=True).apply(np.nanmin)
-        anomalies = np.abs(rollmax - rollmin) < self._threshold
-        anomalies.iloc[0] = False  # first element cannot be determined
-        anomalies.iloc[-1] = False
-        idx = np.where(anomalies)[0]
-        if idx is not None:
-            # assuming window size = 3
-            # remove also points before and after each detected anomaly
-            anomalies.iloc[idx[idx > 0] - 1] = True
-            maxidx = len(anomalies) - 1
-            anomalies.iloc[idx[idx < maxidx] + 1] = True
-
-        return anomalies
+    def _detect(self, data:pd.Series) -> pd.Series:
+        """Detect constant values in single column or multiple columns."""
+        
+        if isinstance(data, pd.DataFrame):
```

Comment:
Please elaborate on this functionality in the PR description. And note that is inconsistent with the type hint, which only expects a Series.